service/history/tasks/predicates.go

```diff
@@ -174,6 +174,52 @@ func (t *TypePredicate) Equals(predicate Predicate) bool {
 	return maps.Equal(t.Types, typePrediate.Types)
 }
 
+// OutboundTaskGroupNamespaceIDAndDestination is the key for grouping tasks by task type namespace ID and destination.
+type OutboundTaskGroupNamespaceIDAndDestination struct {
```

Comment:
nit: guess we don't really need "Outbound" in the name

Reply:
Sure, I can change that.

Reply:
Done.